azurerm/internal/services/appplatform/resource_arm_spring_cloud_service.go

```diff
@@ -0,0 +1,591 @@
+	}
+
+	d.Set("name", resp.Name)
+	d.Set("resource_group_name", id.ResourceGroup)
+	if location := resp.Location; location != nil {
+		d.Set("location", azure.NormalizeLocation(*location))
+	}
+
+	if resp.Properties != nil && resp.Properties.ConfigServerProperties != nil && resp.Properties.ConfigServerProperties.ConfigServer != nil {
+		if props := resp.Properties.ConfigServerProperties.ConfigServer.GitProperty; props != nil {
+			if err := d.Set("config_server_git_setting", flattenArmSpringCloudConfigServerGitProperty(props, d)); err != nil {
+				return fmt.Errorf("failure setting AzureRM Spring Cloud Service error: %+v", err)
+			}
+		}
+	}
+
+	return tags.FlattenAndSet(d, resp.Tags)
+}
+
+func resourceArmSpringCloudServiceDelete(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).AppPlatform.ServicesClient
+	ctx, cancel := timeouts.ForDelete(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := parse.SpringCloudServiceID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	future, err := client.Delete(ctx, id.ResourceGroup, id.Name)
+	if err != nil {
+		return fmt.Errorf("failure deleting Spring Cloud Service %q (Resource Group %q): %+v", id.Name, id.ResourceGroup, err)
+	}
+
+	if err = future.WaitForCompletionRef(ctx, client.Client); err != nil {
+		if !response.WasNotFound(future.Response()) {
+			return fmt.Errorf("failure waiting for deleting Spring Cloud Service %q (Resource Group %q): %+v", id.Name, id.ResourceGroup, err)
+		}
+	}
+
+	return nil
+}
+
+func expandArmSpringCloudConfigServerGitProperty(input []interface{}) *appplatform.ConfigServerGitProperty {
+	if len(input) == 0 || input[0] == nil {
+		return nil
+	}
+
+	v := input[0].(map[string]interface{})
+	if v == nil {
+		return nil
+	}
+
+	result := appplatform.ConfigServerGitProperty{
+		URI: utils.String(v["uri"].(string)),
+	}
+
+	if label := v["label"].(string); label != "" {
+		result.Label = utils.String(label)
+	}
+	if searchPaths := v["search_paths"].([]interface{}); len(searchPaths) > 0 {
+		result.SearchPaths = utils.ExpandStringSlice(searchPaths)
+	}
+	if httpBasicAuth := v["http_basic_auth"].([]interface{}); len(httpBasicAuth) > 0 {
+		v := httpBasicAuth[0].(map[string]interface{})
+		result.Username = utils.String(v["username"].(string))
+		result.Password = utils.String(v["password"].(string))
+	}
+	if sshAuth := v["ssh_auth"].([]interface{}); len(sshAuth) > 0 {
+		v := sshAuth[0].(map[string]interface{})
+		result.PrivateKey = utils.String(v["private_key"].(string))
+		result.StrictHostKeyChecking = utils.Bool(v["strict_host_key_checking_enabled"].(bool))
+
+		if hostKey := v["host_key"].(string); hostKey != "" {
+			result.HostKey = utils.String(hostKey)
+		}
+		if hostKeyAlgorithm := v["host_key_algorithm"].(string); hostKeyAlgorithm != "" {
+			result.HostKeyAlgorithm = utils.String(hostKeyAlgorithm)
+		}
+	}
```

Comment:
We should test against these both being set and error out to the user to prevent both being attempted to be sent to the API since only one will be accepted (based on your comments from the service team that this is conditional based on the protocol used in the URI).

Reply:
I have added check to prevent setting both the fields, please have a look